apps/web-staking/src/server/services/Pool.service.ts

```diff
@@ -152,6 +152,10 @@ export function mapPool(pool: IPool): PoolInfo {
 	 *
 	 * @dev DB Socials structure: [website: string, twitter: string, discord: string, telegram: string, instagram: string, tiktok: string, youtube: string]
 	 */
+	const ownerShare = pool.pendingShares ? pool.pendingShares[0] : 0;
+	const keyBucketShare = pool.pendingShares ? pool.pendingShares[1] : 0;
+	const stakedBucketShare = pool.pendingShares ? pool.pendingShares[2] : 0;
```

Comment:
This is not correct. The pendingShares will only be set when `updateSharesTimestamp` is in the past.

In the StakingPool.sol the shares will be overwritten at the next user interaction, if no user interacts the smart contract will not update the pending shared automatically, thats why we have to do that check here as well.

Reply:
I believe this should be resolved now. Thank you for the guidance!